clientv3/client.go

```diff
@@ -119,6 +119,13 @@ func (c *Client) SetEndpoints(eps ...string) {
 	c.balancer.updateAddrs(eps)
 }
 
+// TrySwitchEndpoint try to make balancer change it's fix endpoint.
+// It may do nothing while their is only one endpoint, or switch has just happened
+// It is called when the endpoint now is not available, like network partition happens
+func (c *Client) TrySwitchEndpoint() {
```

Comment:
I don't think this should be a public interface; the user shouldn't have to give hints to the client to switch endpoints. Just call `c.balancer.trySwitchEndpoint` in `retry.go`?